meson_options.txt

```diff
@@ -30,4 +30,6 @@ option('log_level', type: 'combo', choices: ['trace', 'debug', 'info', 'warning'
 option('rust', type: 'boolean', value: false, description: 'Build Rust bindings')
 
 # Tests
+option('build_tests', type: 'boolean', value: true, description: 'Build all tests')
+option('build_examples', type: 'boolean', value: true, description: 'Build all examples')
```

Comment:
>> We need options that allow us to build examples and test applications.
>> The default value of these options should be false.
>>
>> Tests and examples should be enabled for CI.

> I can add the option but I think they should be enabled by default. This would simplify build for customers to run/try NIXL tests. It would also avoid CI changes. The build time added should be minimal.

The default build usually provides the minimum set of binary files required to use the library. Examples are optional part that helps users write applications using the API provided by the library. I think it would be better to provide an alias/option to build a developer version that includes tests and examples.